File: Frontend/src/app/models/medida.ts
import { Sensor } from "./sensor";

export class Medida {

    public id: number;
    public createdAt: Date;
    public saturacion: number;
    public temperatura: number;

    public sensor: Sensor;

    public constructor(json?: any) {

        Object.assign(this, json);

        if (this.createdAt) { this.createdAt = new Date(this.createdAt) };

    }

    public getJSON(): any {
        return {
            id: this.id,
            createdAt: this.createdAt,
            saturacion: this.saturacion,
            temperatura: this.temperatura,
            sensor: this.sensor != null ? this.sensor.getJSON() : null
        }
    }

}

File: Backend/src/main/java/aqct/backend/model/Medida.java
package aqct.backend.model;

import java.io.Serializable;
import java.sql.Timestamp;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import org.hibernate.annotations.CreationTimestamp;

@Entity
@Table(name="medida")
public class Medida implements Serializable {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;
    
    @Column(name="saturacion")
    private double saturacion;
    
    @Column(name="temperatura")
    private double temperatura;
    
    @Column(name="created_at")
    @CreationTimestamp
    private Timestamp createdAt;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "sensor_id")
    private Sensor sensor;
    
    public Medida() {
        
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public double getSaturacion() {
        return saturacion;
    }

    public void setSaturacion(double saturacion) {
        this.saturacion = saturacion;
    }

    public double getTemperatura() {
        return temperatura;
    }

    public void setTemperatura(double temperatura) {
        this.temperatura = temperatura;
    }

    public Timestamp getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(Timestamp createdAt) {
        this.createdAt = createdAt;
    }

    public Sensor getSensor() {
        return sensor;
    }

    public void setSensor(Sensor sensor) {
        this.sensor = sensor;
    }
    
}
